std/path.d

```diff
@@ -3902,6 +3902,58 @@ if (isConvertibleToString!Range)
     assert(testAliasedString!isValidPath("/foo/bar"));
 }
 
+/** Checks whether `path1` is a subpath of `path2`
+  *
+  * Params:
+  *     path1 = string or Range of characters
+  *     path2 = string or Range of characters
+  *
+  * Returns:
+  *     true if `path` is a subpath of `path2`
+  *     false otherwise
+*/
+bool isSubPath(Range)(Range subPath, Range path)
```

Comment:
May I suggest the name `pathStartsWith`, for consistency with `std.algorithm.startsWith`.

The order of the arguments would then be inverted.

Or, is there a reason why this makes more sense that I'm not seeing?